### DISCUSSION TITLE
npx signal forwarding isn't working as expected in Kubernetes

### DISCUSSION THREAD

**Discussion by madhusudancs** [community/community#141011] [Programming Help]
We launch our Nodejs services as Kubernetes. We have an ENTRYPOINT command in our Dockerfile that looks like this:

```
ENTRYPOINT ["npx", "cross-env", "NODE_ENV=production", "node",  "--import", "./build/common/instrumentation.js", "./build/service/src/index.js"]
```

When the pod that runs this container is killed, service process launched as `./build/service/src/index.js` never gets called on its `process.on('SIGTERM', ... )`

Using `[killsoop](https://github.com/iovisor/bcc/blob/master/tools/killsnoop.py)` from `bcc-tools` it is clear that SIGTERM never gets forwarded beyond a `sh -c` process that's spawned. Here is the process tree for our pod

```
npm exec cross-,1
  |-sh,17 -c cross-env NODE_ENV=production node --import ./build/common/instrumentation.js ./build/service/src/index.js
  |   `-node,18 /myapp/node_modules/.bin/cross-env NODE_ENV=production node --import ./build/common/instrumentation.js ./build/service/src/index.js
  |       |-node,25 --import ./build/common/instrumentation.js ./build/service/src/index.js
  |       |   |-{node},26
  |       |   |-{node},27
  |       |   |-{node},28
```

Here is the killsnoop output:
```
$ ./killsnoop
TIME      PID    COMM             SIG  TPID   RESULT
...
19:06:08  588834 runc             15   521532 0
19:06:08  521532 npm exec cross-  15   17     0
19:06:08  521532 npm exec cross-  15   1      0
19:06:09  588869 runc             9    521323 0
```

Does npm exec create a wrapper shell process (with `sh -c`)? How does this work? Any idea why signal forwarding wouldn't work with our setup?

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @madhusudancs ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Reply by madhusudancs** [reply]
Thanks! Made a copy of this as the original post that was moved was closed.

**Reply by gmile** [reply]
@madhusudancs did the discussion continue elsewhere? Were you able to find a solution to this? I suspect this could be `npx` issue 🤔
